lib/Request.js

```diff
@@ -124,101 +133,124 @@ Request.prototype.setQuery = function(query, value) {
     return this;
 };
 
+
+Request.prototype.deleteQuery = function(query) {
+    delete this._url.query[query];
+    return this;
+};
+
 Request.prototype.getQuery = function(query) {
     return this._url.query.hasOwnProperty(query) ? this._url.query[query] : undefined;
 };
 
+Request.prototype.validateAndCastParams = function(object) {
+    return this.validateAndCast(this.getParams());
+}
+
+Request.prototype.validateAndCastQueries = function(object) {
+    return this.validateAndCast(this.getQueries());
+
+}
+
 /**
- * Validates the query with the provided schema.
+ * Validates the query  or passed in params with the provided schema.
  * Query parameters that are not included in the schema are ignored.
- * Query parameters that are defined as type "integer" or "number" are converted accordingly.
- * Returns a validation object with two properties:
- *   - boolean valid -- true if the query parameters are valid
- *   - array errors -- array of properties that are in error, if any
+ * Query parameters that are defined as type "integer", "number", "boolean" or "string" are converted accordingly.
+ * Pushes errors to validationErrors array if errors are gound
  *
- * @param object schema - a schema in json-schema format.
- * @return object - validation result
+ * @param object - object to validate
  */
-Request.prototype.validateQuery = function(schema, defaults) {
-    // Nothing to validate if the schema is not an object (rare)
-    if (!schema.properties) {
-        // Clear out existing query params -- they must be in the schema to access them.
-        this.setQueries({});
-        return {
-            valid: true,
-            errors: []
-        };
+Request.prototype.validateAndCast = function(object) {
+    if (!Object.keys(object).length) {
```

Comment:
Use [`Object.getOwnPropertyNames(object)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyNames) to ignore properties on the prototype chain.